airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java

```diff
@@ -140,6 +140,18 @@ public Optional<JobRead> getLatestSyncJob(final UUID connectionId) throws IOExce
     return jobPersistence.getLastSyncJob(connectionId).map(JobConverter::getJobRead);
   }
 
+  public List<Optional<JobRead>> getLatestSyncJobsForConnections(final List<UUID> connectionIds) throws IOException {
```

Comment:
curious, why use a list of optionals here? If the list is empty, that suggest we don't have any items to return. Feels like we can simplify the type signature here.